### PR TITLE
feat!: multiple services depending on same plugin implementation

### PR DESCRIPTION
**Pull request by leoporoli**

### CODE REVIEW

File: ci/obd-demo.yaml [hunk 1]
```diff
@@ -332,3 +299,51 @@ spec:
...
                   number: 80
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: jsdelivr-api
+  annotations:
+    kardinal.dev.service/plugin-definition: |
+      - name: github.com/leoporoli/jsdelivr-api-plugin
```
Comment (leoporoli) [inline ci/obd-demo.yaml:310]: will replace it in a next PR after merging plugins PRs

File: ci/obd-demo.yaml [hunk 2]
```diff
@@ -332,3 +299,51 @@ spec:
...
+          api_key: "dev"
+
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: postgres-seed-plugin
+  annotations:
+    kardinal.dev.service/plugin-definition: |
+      - name: github.com/leoporoli/postgres-seed-plugin
```
Comment (leoporoli) [inline ci/obd-demo.yaml:323]: will replace it in a next PR after merging plugins PRs

File: kontrol-service/engine/flow/dev_flow.go [hunk 3]
```diff
@@ -141,228 +132,282 @@ func markParentsAsShared(topology *resolved.ClusterTopology, service *resolved.S
...
+		}
 
-			// Apply a chain of stateful plugins to the stateful service
-			resultSpec := modifiedService.WorkloadSpec.DeepCopy()
-			for pluginIdx, plugin := range modifiedService.StatefulPlugins {
-				if plugin.Type == "external" {
-					// we handle external plugins above
-					// might need to handle this if stateful services can have external dependencies
-					continue
+		// TODO SECTION 2 - Target service updates with new modifications
```
Comment (tedim52) [inline kontrol-service/engine/flow/dev_flow.go:223]: is the TODO still needed here?

Comment (leoporoli) [reply kontrol-service/engine/flow/dev_flow.go]: nope, will remove it

**Review by tedim52 — COMMENTED** [review]
LGTM! I think separating out the plugins into their own definition simplified the code a lot!

**Review by tedim52 — APPROVED** [review]

**Pull request merged**